log/handler.go

```diff
@@ -383,31 +392,34 @@ func BufferedHandler(bufSize int, h Handler) Handler {
 // it if you write your own Handler.
 func LazyHandler(h Handler) Handler {
 	return FuncHandler(func(r *Record) error {
-		// go through the values (odd indices) and reassign
-		// the values of any lazy fn to the result of its execution
-		hadErr := false
-		for i := 1; i < len(r.Ctx); i += 2 {
-			lz, ok := r.Ctx[i].(Lazy)
-			if ok {
-				v, err := evaluateLazy(lz)
-				if err != nil {
-					hadErr = true
-					r.Ctx[i] = err
-				} else {
-					if cs, ok := v.(stack.CallStack); ok {
-						v = cs.TrimBelow(r.Call).TrimRuntime()
-					}
-					r.Ctx[i] = v
+		lazyEvaluateRecord(r)
+		return h.Log(r)
+	})
+}
+
+func lazyEvaluateRecord(r *Record) {
+	// go through the values (odd indices) and reassign
+	// the values of any lazy fn to the result of its execution
+	hadErr := false
+	for i := 1; i < len(r.Ctx); i += 2 {
+		lz, ok := r.Ctx[i].(Lazy)
```

Comment:
Nitpick, maybe merge the two, like `if lz, ok := r.Ctx[i].(Lazy); ok {`
Otherwise LGTM, thanks!

Reply:
Actually, I didn't change code here, just extract code from the original `LazyHandler` so that it can be re-used.

Reply:
I see, it just caught my eye :) Could you please do this change in order to keep the code more compact?

Reply:
Sure!